server/command.go

```diff
@@ -30,6 +31,13 @@ type Handler interface {
 	HandleOtherCommand(cmd byte, data []byte) error
 }
 
+type ReplicationHandler interface {
+	//handle Replication command
+	HandleRegisterSlave(data []byte) error
+	HandleBinlogDump(pos *Position, s *replication.BinlogStreamer)
+	HandleBinlogDumpGTID(gtidSet *MysqlGTIDSet, s *replication.BinlogStreamer)
```

Comment:
I think that `s *replication.BinlogStreamer` shouldn't be here. It is up to users who will implement `ReplicationHandler` what to do with these events.

Reply:
Sorry I don't understand, you mean we can expose `Conn` to user and user directly write binlog event as []byte to `Conn`?

Reply:
Oh... I worded it really badly.

I will agree with you that it is not a good idea to expose `Conn`.

Reply:
For consistency we can
* move `s *replication.BinlogStreamer` to return values
* add `error` to return values
* remove `go` in `Conn.dispatch()`... or document that this method will be called form its own coroutine.